students/jim_krecek/lesson04/mailroom_p2.py

```diff
@@ -10,15 +10,13 @@ def send_ty():
         for d in data.keys():
             print(d)
     else:
-        for d in data.keys():
-            if askname.lower() == d.lower():
-                askmoney = input("\nHow much did this person donate?\n>>> ")
-                data[d].append(int(askmoney))
-                emailtext = "\nDear {},\n\nThank you for your generous donation of ${}.\n\nSincerely, Jim".format(d,askmoney)
-                print(emailtext)
-                break
-    
-        if askname != d:
+        if data.get(askname):
+            askmoney = input("\nHow much did this person donate?\n>>> ")
+            data[askname].append(int(askmoney))
+            emailtext = "\nDear {},\n\nThank you for your generous donation of ${}.\n\nSincerely, Jim".format(askname,askmoney)
+            print(emailtext)
+
+        else:
             askmoney = input("\nHow much did this person donate?\n>>> ")
```

Comment:
Much better! 

Something I would recommend improving maybe in the next mailroom assignment is some of the duplication.
It's one of the best practices to remove code that is obviously duplicated. Right lines 13 through 24 share a lot of duplicate code and can be fairly easily addressed 🙂 

https://refactoring.guru/smells/duplicate-code

Reply:
touche! I definitely see some lines that could be moved in that block of code. thanks for pointing that out.